### ISSUE TITLE
javax.ws transitive dependency resolution issue with sbt

### ISSUE THREAD

**Issue by ncolomer** [DataDog/datadog-api-client-java#481]
**Describe the bug**
When you reference `datadog-api-client` as a dependency in a scala sbt project (at least with sbt 1.2.8), you get an issue when it comes to resolve `javax.ws.rs:javax.ws.rs-api`, pulled by library's explicit dependency `org.glassfish.jersey.core:jersey-client`.

The problem is caused by sbt and is largely discussed in https://github.com/sbt/sbt/issues/3618.

Amongst all the workarounds mentioned in the thread above, the only one that solved the problem for me was to replace `javax.ws.rs:javax.ws.rs-api` by `jakarta.ws.rs:jakarta.ws.rs-api`:

```scala
libraryDependencies ++= Seq(
  ...
  "com.datadoghq" % "datadog-api-client" % "1.0.0-beta.6" exclude("javax.ws.rs", "javax.ws.rs-api"),
  "jakarta.ws.rs" % "jakarta.ws.rs-api" % "2.1.2", // see https://github.com/sbt/sbt/issues/3618
```

Maybe a good way to "workaround" the problem would be to use another high level http client library.

**To Reproduce**
1. create a scala sbt project
2. add `"com.datadoghq" % "datadog-api-client" % "1.0.0-beta.6"` as a dependency
3. run `sbt update`
4. see error:
```
[warn] 	Detected merged artifact: [FAILED     ] javax.ws.rs#javax.ws.rs-api;2.1!javax.ws.rs-api.${packaging.type}:  (0ms).
[warn] ==== local: tried
[warn] ==== public: tried
[warn]   https://repo1.maven.org/maven2/javax/ws/rs/javax.ws.rs-api/2.1/javax.ws.rs-api-2.1.${packaging.type}
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::              FAILED DOWNLOADS            ::
[warn] 	:: ^ see resolution messages for details  ^ ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: javax.ws.rs#javax.ws.rs-api;2.1!javax.ws.rs-api.${packaging.type}
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[error] sbt.librarymanagement.ResolveException: download failed: javax.ws.rs#javax.ws.rs-api;2.1!javax.ws.rs-api.${packaging.type}
[error] 	at ...
[error] (update) sbt.librarymanagement.ResolveException: download failed: javax.ws.rs#javax.ws.rs-api;2.1!javax.ws.rs-api.${packaging.type}
```

**Expected behavior**
No dependency download issue when installing the `datadog-api-client` library in your project.

**Environment and Versions (please complete the following information):**
Using `com.datadoghq:datadog-api-client:1.0.0-beta.6` (`1.0.0-beta.7` is not available in any public repo).

**Comment by michael-treadwell** [comment]
I am still facing this issue as of 2022-02-22 and the workaround posted above worked for me.

**Comment by therve** [comment]
We released 1.6.0 with a shaded jar, which should alleviate these sorts of issues. Thanks.
